client/src/Components/Common/CommentContainer/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+import { List } from 'antd';
+import PropTypes from 'prop-types';
+import Comment from '../Comment';
+import MainButton from '../MainButton';
+
+const commentsPerPage = 5;
+let arrayForHoldingComments = [];
+
+const CommentContainer = ({ data }) => {
+  const [commentsToShow, setCommentsToShow] = useState([]);
+  const [next, setNext] = useState(5);
+
+  const loopWithSlice = (start, end) => {
+    const slicedComments = data.slice(start, end);
+    arrayForHoldingComments = [...arrayForHoldingComments, ...slicedComments];
+    setCommentsToShow(arrayForHoldingComments);
+  };
+
+  const handleShowMoreComments = () => {
+    loopWithSlice(next, next + commentsPerPage);
+    setNext(next + commentsPerPage);
+  };
+
+  useEffect(() => {
+    loopWithSlice(0, commentsPerPage);
+  }, []);
+
+  return (
+    <div>
+      <List
+        itemLayout="horizontal"
+        dataSource={commentsToShow}
+        renderItem={(item) => (
+          <List.Item>
+            <Comment
+              isAdmin={item.isAdmin}
+              userName={item.userName}
+              commentText={item.commentText}
+              date={item.date}
+              rateValue={item.rateValue}
+            />
+          </List.Item>
+        )}
+      />
+      {commentsToShow.length !== data.length ? (
+        <MainButton onClick={handleShowMoreComments}>
+          عرض المزيد من التعليقات
+        </MainButton>
+      ) : (
+        <MainButton backgroundColor="#ABC4C1">
+          عرض المزيد من التعليقات
+        </MainButton>
+      )}
+    </div>
+  );
+};
+CommentContainer.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      isAdmin: PropTypes.bool,
+      userName: PropTypes.string,
+      commentText: PropTypes.string,
+      date: PropTypes.string,
+      rateValue: PropTypes.number,
+    })
+  ).isRequired,
```

Comment:
you are ( or the antd list ) handdling the case if there is no data comes and disply this : 
![image](https://user-images.githubusercontent.com/39485495/107155478-47b9f900-6981-11eb-8ecb-6511c808303b.png)

but throw an error in the console because the `data` is typed as required  in  the component you can avoid that by passing default prop `data` as an empty array and remove is required 
```suggestion
CommentContainer.defaultProps = {
  data : []
}

CommentContainer.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      isAdmin: PropTypes.bool,
      userName: PropTypes.string,
      commentText: PropTypes.string,
      date: PropTypes.string,
      rateValue: PropTypes.number,
    })
  ),
```